plan/relations.go

```diff
@@ -1566,30 +1566,80 @@ func (s *SetRel) Remap(mapping ...int32) (Rel, error) {
 	return RemapHelper(s, mapping)
 }
 
+// ExtensionRelDefinition provides flexibility for extension relations to define
+// their own schema and expression handling behavior.
+type ExtensionRelDefinition interface {
+	// Schema returns the output schema for this extension given the input relations.
+	Schema(inputs []Rel) types.RecordType
+
+	// Build returns the protobuf Any type containing the extension details.
+	Build(inputs []Rel) *anypb.Any
+
+	// Expressions returns any expression trees within this extension.
+	// This is optional and can return nil if the extension contains no expressions.
+	Expressions(inputs []Rel) []expr.Expression
+}
+
+// UnknownExtension wraps an *anypb.Any detail for extension relations parsed from protobuf that don't have a definition.
+type UnknownExtension struct {
+	detail *anypb.Any
+}
+
+// Schema returns an empty record type for unknown extensions.
+func (ue *UnknownExtension) Schema(inputs []Rel) types.RecordType {
+	if len(inputs) > 0 {
+		return inputs[0].RecordType()
+	}
```

Comment:
This doesn't match the docstring above. And technically, for backwards-compatibility, this should only trigger for `len(inputs) == 1`, right? (well, technically, that would *still* be wrong for an `ExtensionMultiRel` with 1 input, but I think would be close enough…)

My suggestion: let's get it a bit closer to the previous behavior, and note the situation

```suggestion
	if len(inputs) == 1 {
		// XXX: the UnknownExtension doesn't know its schema type. For now, we
		// guess that if there is exactly one input, its probably that record
		// type, otherwise, empty; that's not a very good choice, but its now
		// what we have backwards-compatibility-wise. We should probably hook
		// this into the ExtensionRegistry or similar so we can look it up and
		// get the correct schema.
		return inputs[0].RecordType()
	}
```